website_chd_product_configurator/controllers/controllers.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+from openerp import http
+from openerp.http import request
+from openerp.osv import orm
+import json
+
+
+class Chd_website(http.Controller):
+
+    @http.route('/chd_init/', auth='public', website=True)
+    def start(self, selected_id=False, type=False):
+        partner = self.get_current_partner()
+        if partner is False:
+            return request.render(
+                'website_chd_product_configurator.no_partner', {})
+        Conf_products = request.env['product.template']
+        accessories = request.env['product.product']
+        chd_price_components_at = request.env[
+            'price.component.attribute.template']
+        chd_types = request.env['product.type']
+        chd_finishing = request.env['product.finishing']
+        # when posting a selected product to configure
+        if request.httprequest.method == 'POST' and selected_id:
+            curr_types = chd_types.search([
+                ('product_option_ids', 'in', [selected_id])
+                ])
+            curr_product = Conf_products.search([('id', '=', selected_id)])
+            curr_chd_price_component_ats = chd_price_components_at.search([
+                ('id', 'in', curr_product.attribute_template_ids.ids),
+                ('active', '=', True)
+                ])
+            avail_accessories = accessories.search([
+                ('id', 'in', curr_product.chd_accessoire_ids.ids)
+                ])
+            return request.render(
+                'website_chd_product_configurator.configurator',
+                {
+                    'curr_product_id': curr_product,
+                    'curr_types': curr_types,
+                    'curr_chd_price_component_ats':
+                    curr_chd_price_component_ats,
+                    'avail_accessories': avail_accessories,
+                    })
+        # first iteration, loading the page
+        return request.render('website_chd_product_configurator.conf_start', {
+            'conf_products': Conf_products.search([(
+                'chd_origin_product', '=', True)
+                ]),
+            })
+
+    # shows the options list
+    @http.route('/chd_init/<id>/', website=True)
```

Comment:
If you know what you expect, tell werkzeug: `<int:id>` - then it will do validation and type casting for you